packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -146,41 +146,57 @@ export class EngineReportingExtension<TContext = any>
       }
     }
 
-    if (this.options.privateVariables !== true && o.variables) {
-      // Note: we explicitly do *not* include the details.rawQuery field. The
-      // Engine web app currently does nothing with this other than store it in
-      // the database and offer it up via its GraphQL API, and sending it means
-      // that using calculateSignature to hide sensitive data in the query
-      // string is ineffective.
-      this.trace.details = new Trace.Details();
-      Object.keys(o.variables).forEach(name => {
-        if (
-          this.options.privateVariables &&
-          Array.isArray(this.options.privateVariables) &&
-          // We assume that most users will have only a few private variables,
-          // or will just set privateVariables to true; we can change this
-          // linear-time operation if it causes real performance issues.
-          this.options.privateVariables.includes(name)
-        ) {
-          // Special case for private variables. Note that this is a different
-          // representation from a variable containing the empty string, as that
-          // will be sent as '""'.
-          this.trace.details!.variablesJson![name] = '';
-        } else {
-          try {
-            this.trace.details!.variablesJson![name] = JSON.stringify(
-              o.variables![name],
-            );
-          } catch (e) {
-            // This probably means that the value contains a circular reference,
-            // causing `JSON.stringify()` to throw a TypeError:
-            // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Issue_with_JSON.stringify()_when_serializing_circular_references
-            this.trace.details!.variablesJson![name] = JSON.stringify(
-              '[Unable to convert value to JSON]',
-            );
+    if (
+      (this.options.enforcePrivateVariables ||
+        this.options.privateVariables === null) &&
+      o.variables
+    ) {
+      // Default to the enforcePrivateVariables specs if privateVariables was not set.
+      this.trace.details = makeTraceDetails(
+        o.variables,
+        this.options.enforcePrivateVariables,
+        o.queryString,
+      );
+    } else {
+      // [TO BE DEPRECATED] to use the parameter enforcePrivateVariables over privateVariables.
+      // If privateVariables was previously specified (non-null), trace.details.variablesJson will
+      // look the same as before.
+      if (this.options.privateVariables !== true && o.variables) {
```

Comment:
I'm not sure why it makes sense to handle the new option in makeTraceDetails and the old option outside of it. Seems like this means your unit tests are not actually testing the full behavior here (which includes the possibility of hitting this else clause). It seems like either both should be handled in makeTraceDetails, or there should be two functions, makeTraceDetails and makeTraceDetailsLegacy or something.  (I guess the answer is "because you only needed to unit test one of them.)

Reply:
I separated these mostly because I didn't want to change the old logic/behavior, where if `privateVariables === true` then none of the variable keys would be recorded in the trace details (and I assumed the old option was working as expected, so I didn't write test cases for it-- my bad!). I'll wrap the old into another function, and do a couple more simple test cases.

Reply:
Another thought is that, if there really are just two ways of specifying equivalent options, you could normalize to the new version in the EngineReportingExtension constructor.

Reply:
I think that normalizing to the new version makes the most sense, but would it be okay to change the output of the `privateVariables === true` then? I was worried I would change the expected output for some users.. not sure how this might be being used.

Reply:
I'm not sure what you mean specifically. But I also don't understand why you're changing the option name at all?

Reply:
I was referring to when `privateVariables === true` right now, then all variable keys themselves will be hidden (nothing shows up in the Query Variables or Private Variables fields in the Trace Inspector UI), sorry for not clarifying! We talked about this not really making sense, and wanting to still have all the variable keys listed in the Private Variables box in the UI. I was planning on addressing this issue within the new option and not changing how `privateVariables` currently works, because I wasn't sure if any users were relying on the existing behavior.

But RE the option name -- I saw your other comment, will respond to you there!

Reply:
Ah right, I forgot that distinction. I *think* it's not actually a very useful distinction: in general for a valid query, all variable names are listed in the query itself. (You actually can include variables that aren't part of the query (see https://github.com/graphql/graphql-spec/issues/540) but that shouldn't be common.) I think "need to hide variable names" is an obscure enough requirement that people who want it can use a custom function instead of having an easy boolean (or having it be the default).

Reply:
Addressed this, so that both the old and new options are handled by the same code block.